sonic-net/sonic-buildimage
# [Edgecore][sonic-platform-modules-accton]Support kernel 6.1 and bookworm

#### Why I did it
The sonic-platform-modules-accton package builds were failing to build against the 6.1 kernel in bookworm branch.
##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Enable rule.mk to let build sonic-platform-modules-accton
Modify accton driver that use "i2c_driver->remove callback function" to use "void" instead of "init".
Because kernel 6.1 has change to be below.

struct i2c_driver {
	unsigned int class;

	/* Standard driver model interfaces */
	int (*probe)(struct i2c_client *client);
	void (*remove)(struct i2c_client *client);

Old kernel define as below,
 .....
init (*remove)(struct i2c_client *client);
#### How to verify it
Build deb.
BLDENV=bookworm make -f Makefile.work $@
Check deb can build successfully


#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
